toucan_connectors/toucan_connector.py

```diff
@@ -504,3 +505,68 @@ def format_db_model(
             .reset_index()
             .to_dict('records')
         )
+
+
+class MalformattedVersion(Exception):
+    """raise when the given version of the engine is not well formated"""
+
+
+class UnavailableVersion(Exception):
+    """raise when the version of the engine is not available"""
+
+
+class VersionableEngineConnector(ABC):
+    """
+    This class is responsable for formating and rendering the version engine
+    from which the connector is fetching data from !
+
+    We suppose all versions should fit the semver format and for that, we
```

Comment:
During the meeting, we said that we do **not** suppose that all versions are semver. Matching logic looks good though :slightly_smiling_face: